compiler/rustc_builtin_macros/src/deriving/generic/ty.rs

```diff
@@ -245,7 +237,15 @@ impl Bounds {
             })
             .collect();
 
-        mk_generics(generic_params, span)
+        Generics {
+            params,
+            where_clause: ast::WhereClause {
+                has_where_token: false,
+                predicates: Vec::new(),
+                span: self_generics.where_clause.span,
+            },
+            span: self_generics.span,
```

Comment:
This function is used for 1) building generics for generated impls and 2) building generics for methods inside these generated impls.
`self_generics` are generics on the derive's target ADT (struct or enum).
In the second case the spans have no relation to the ADT's generics spans, so these spans should either be passed from the outside of `to_generics`, or set outside of `to_generics`, like `params` and `predicates`.

Reply:
Changed.